src/psbt.h

```diff
@@ -548,8 +550,36 @@ struct PartiallySignedTransaction
     }
 };
 
+enum class PSBTRole {
+    UPDATER,
+    SIGNER,
+    FINALIZER,
+    EXTRACTOR
+};
+
+struct PSBTInputAnalysis {
```

Comment:
It'd be nice to have some documentation for these structs. There could be a comment saying struct fields correspond to fields returned by the `analyzepsbt` method, if you didn't want to just copy the documentation from `analyzepsbt`

Reply:
Nah, thanks, this is a good point, I may as well document them properly.